### PR TITLE
fix(matchers): visibility matchers work with Jest

### LINKED ISSUE #97: feat: make toBeVisible take into account 'hidden' attribute
Hey folks,
I noticed `toBeVisible` matcher ignore [`hidden` attribute](https://www.w3schools.com/tags/att_global_hidden.asp).
It may be a good idea to check that too.

### PR DESCRIPTION
**Pull request by IlCallo**
I recreated the visibility check to be jsdom friendly to make them work with Jest too.
Mixes jQuery 1 and 3 :visible and :hidden pseudo-selector implementation, because jQuery 3 implementation won't work in jsdom: layout isn't managed.

Also checks for 'hidden' attribute.

(I accidentally closed previous PR while force pushing an update)

Fixes #97

### CODE REVIEW

File: projects/spectator/src/lib/matchers.ts [hunk 1]
```diff
@@ -240,18 +240,42 @@ export const toBeEmpty = comparator(el => {
...
+ * Hidden elements are elements that have:
+ * 1. Display property set to "none"
  * 2. Width and height set to 0
  * 3. A hidden parent element (this also hides child elements)
- * 4. Form elements with type="hidden"
+ * 4. Type equal to "hidden" (only for form elements)
+ * 5. A "hidden" attribute
+ */
+function isHidden(el) {
+  while (el) {
```
Comment (NetanelBasal) [inline projects/spectator/src/lib/matchers.ts]: Why is the check not only on the element?

Comment (IlCallo) [reply projects/spectator/src/lib/matchers.ts]: You mean, why did I abstracted the isHidden function instead of writing it directly inside the matcher function?
Because it's used both for `toBeVisible` and `toBeHidden` matcher.

Otherwise I didn't understood the question, could you rephrase it?

Comment (NetanelBasal) [reply projects/spectator/src/lib/matchers.ts]: I mean, why traversed the tree? The check should be only on the element itself.

Comment (benelliott) [reply projects/spectator/src/lib/matchers.ts]: @NetanelBasal In the old JSDoc it says

```
 * Hidden elements are elements that are:
...
 * 3. A hidden parent element (this also hides child elements)
```

Which is probably taken from https://api.jquery.com/hidden-selector/

So the previous jQuery implementation was probably using this methodology too.

Comment (NetanelBasal) [reply projects/spectator/src/lib/matchers.ts]: Yes, I know. I meant, do we really need it in a spec context? Usually I want to test the exact element.

Comment (benelliott) [reply projects/spectator/src/lib/matchers.ts]: Dunno, but changing it will probably need a major version bump.

Comment (NetanelBasal) [reply projects/spectator/src/lib/matchers.ts]: Yes, let's leave it like this.
